--- Canna.conf.orig	2026-05-09 23:08:44
+++ Canna.conf	2026-06-25 10:52:23
@@ -108,7 +108,7 @@
 # define DefErrDir $(cannaPrefix)/log
 # define DefLibCannaDir $(cannaExecPrefix)/lib
 #else 
-# define DefCannaPrefix /usr/local/canna
+# define DefCannaPrefix __PREFIX__
 # ifdef ObsoleteInstallDir
 #  define DefErrDir /usr/spool/canna
 #  define DefLibCannaDir /usr/lib
@@ -146,12 +146,12 @@
 XCOMM cannaLibDir = /var/lib/canna
 XCOMM cannaLibDir = /var/db/canna
 cannaLibDir = DefCannaLibDir
-XCOMM cannaManDir = $(cannaPrefix)/share/man
-cannaManDir = $(cannaPrefix)/man
+cannaManDir = $(cannaPrefix)/share/man
+XCOMM cannaManDir = $(cannaPrefix)/man
 cannaIncDir = $(cannaPrefix)/include/canna
 
 libCannaDir = DefLibCannaDir
-ErrDir  = DefErrDir
+ErrDir  = ${cannaPrefix}/var/log/canna
 
 /* 旧バージョンとの互換APIを無効にする場合は0と定義してください */
 #define SupportOldWchar 1
@@ -175,8 +175,8 @@
 #endif
 
 
-cannaOwner = bin
-cannaGroup = bin
+cannaOwner = canna
+cannaGroup = canna
 
 #ifdef InstallAsUser
 cannaOwnerGroup =
@@ -211,7 +211,7 @@
 #endif
 
 /* 日本語マニュアルを使わないのであればコメントアウト */
-#define JAPANESEMAN
+/* #define JAPANESEMAN */
 
 /* #define engineSwitch */
 
@@ -243,7 +243,7 @@
 /* #define UseInstalledLibCanna YES */
 #define UseInstalledLibCanna NO
 
-DicDir   = $(cannaLibDir)/dic
+DicDir   = ${cannaPrefix}/var/db/canna/dic
 
 /* ここから下は変更不要です */
 
@@ -354,7 +354,7 @@
 AccessFile = $(cannaPrefix)/etc/hosts.canna
 #else
 UnixSockDir = /tmp/.iroha_unix
-AccessFile = /etc/hosts.canna
+AccessFile = $(cannaPrefix)/etc/hosts.canna
 #endif
 UnixSockName = IROHA
 
